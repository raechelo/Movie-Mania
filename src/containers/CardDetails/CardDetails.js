import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';

class CardDetails extends Component {
  constructor() {
    super()
    this.state = {
      redirected: false
    }
  }

  handleClick = () => {
    this.setState({ redirected: !this.state.redirected})
  }


    render () {
      const {title, rating, synopsis, posterImage, releaseDate, favorited} = this.props
    const heartClasses = favorited ? 'fas fa-heart fa-2x favorited' : 'fas fa-heart fa-2x';

      return (
        <section className="Card-Details" >
          <h2>{title}</h2>
          <i onClick={this.handleFavoriteClick} className={heartClasses}></i>
          <h5>Released<span>{releaseDate.slice(0, 4)}</span></h5>
          <h6>{rating}/10<i class="far fa-star"></i></h6>
          <p>{synopsis}</p>
          <img src={posterImage} alt={title + ' poster'} />
          <button onClick={this.handleClick} className="back-btn"><i class="fas fa-angle-left"></i> Back to results</button>
          {this.state.redirected && <Redirect to='/' />}
        </section>
      )
  }
}

export default CardDetails;