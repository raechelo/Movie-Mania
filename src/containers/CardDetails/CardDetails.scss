.Card-Details {
  background-color: $bgcolor;
  box-shadow: 2px 2px 2px #3d3d3d;
  display: grid;
  grid-template-areas: "img title title"
  "img date rating"
  "img info info"
  "img btn btn";
  padding: .5rem;
  width: 60%;

  img {
    grid-area: img;
    margin: 1rem;
  }

  h2, h6 {
    color: $accent;
    font-family: $fancyfont;
    text-shadow: 1px 1px 1px lighten($accent, 70%);
  }

  h2 {
    font-size: 4rem;
    grid-area: title;
    height: 3.5rem;
    margin: 0 auto;
  }

  h5 {
    color: $accent;
    font-family: $fancyfont;
    font-size: 1.7rem;
    grid-area: date;
    margin: auto;
    text-align: center;
    text-shadow: 1px 1px 1px lighten($accent, 70%);

    span {
      display: block;
    }
  }

  h6 {
    font-size: 1.5rem;
    grid-area: rating;
    margin: auto auto;
    text-align: center;

    span {
      display: block;
    }
  }

  p {
    color: white;
    font-family: $bodyfont;
    grid-area: info;
    padding: 0 .5rem;
  }

  .rating {
    position: absolute;
  }
}

.back-btn {
  background-color: white;
  color: $accent;
  font-family: $bodyfont;
  font-size: 1rem;
  grid-area: btn;
  margin: auto;
  z-index: 5;

  &:hover {
    background-image: linear-gradient(to left, $accent 50%, white 50%);
    background-size: 200% 100%;
    background-position: right bottom;
    transition: all .8s ease-out;
    color: white;
    cursor: pointer;

    i {
      animation: carat .25s ease-in;
    }
  }

}