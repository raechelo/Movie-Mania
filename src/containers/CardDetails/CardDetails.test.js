import React from 'react';
import { shallow } from 'enzyme';
import CardDetails from './CardDetails';

describe('CardDetails', () => {
  
  let wrapper;
  
  let movie = { title: 'Return of the King', vote_average: 8.9, release_date: "2003-05-10", overview: 'The rightful king retakes the throne of Men and Gondor.'}

  beforeEach(() => {
    wrapper = shallow(<CardDetails {...movie} />)
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();

    movie.favorited = true;
    expect(wrapper).toMatchSnapshot();
  });
})