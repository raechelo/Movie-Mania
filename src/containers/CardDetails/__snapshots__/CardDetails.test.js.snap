// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails should match the snapshot with all the data passed in 1`] = `undefined`;

exports[`CardDetails should match the snapshot with all the data passed in 2`] = `undefined`;
